lib/Conversion/HWToLLVM/HWToLLVM.cpp

```diff
@@ -173,11 +169,11 @@ struct ArrayGetOpConversion : public ConvertOpToLLVMPattern<hw::ArrayGetOp> {
     if (cast<LLVM::LLVMPointerType>(arrPtr.getType()).isOpaque())
       gep = rewriter.create<LLVM::GEPOp>(
           op->getLoc(), LLVM::LLVMPointerType::get(rewriter.getContext()),
-          elemTy, arrPtr, ArrayRef<Value>({zeroC, zextIndex}));
+          elemTy, arrPtr, ArrayRef<LLVM::GEPArg>{zextIndex});
```

Comment:
I think instead of passing `elemTy` (element type of the array), we need to pass the type of the array here (element type of the pointer). That's something I overlooked in the last LLVM bump. Then we also need the zero. This only triggers a verifier when we export to LLVMIR (e.g., when running arcilator).

Reply:
That would make a lot of sense. The GEP documentation and assembly format is a bit confusing. `llvm.getelementptr %0[%1]` looks very much like what you'd want :rofl:, but that's without the zero. I guess that didn't change on the LLVM side and is still needed.

Reply:
Fixed :+1: